CONTRIBUTING.md

```diff
@@ -1,35 +1,77 @@
-
-How to Contribute Code to ROOT
-==============================
+# How to Contribute Code to ROOT
```

Comment:
We should at some point make clear that these are guidelines, as it was done during the ROOT meeting.